custom/enikshay/integrations/bets/repeaters.py

```diff
@@ -0,0 +1,235 @@
+
+    @classmethod
+    def available_for_domain(cls, domain):
+        return BETS_INTEGRATION.enabled(domain)
+
+    def case_types_and_users_allowed(self, case):
+        return self._allowed_case_type(case) and self._allowed_user(case)
+
+
+class BaseBETSVoucherRepeater(BaseBETSRepeater):
+    """Forward a voucher to BETS
+    Case Type: Voucher
+    Trigger: When voucher.state transitions to "approved" or "partially_approved"
+             and voucher.sent_to_bets != 'true'
+    Side Effects:
+        Success: voucher.event_{EVENT_ID} = "true" and voucher.bets_{EVENT_ID}_error = ''
+        Error: voucher.bets_{EVENT_ID}_error = 'error message'
+    """
+    event_id = None
+    voucher_type = None
+
+    def allowed_to_forward(self, voucher_case):
+        if not self.case_types_and_users_allowed(voucher_case):
+            return False
+
+        case_properties = voucher_case.dynamic_case_properties()
+        correct_voucher_type = case_properties['voucher_type'] == self.voucher_type
+        approved = case_properties.get("state") == "approved"
+        not_sent = case_properties.get("event_{}".format(self.event_id)) != "sent"
+        return (
+            approved
+            and correct_voucher_type
+            and not_sent
+            and case_properties_changed(voucher_case, ['state'])
+            and is_valid_voucher_submission(voucher_case)
+        )
+
+
+class ChemistBETSVoucherRepeater(BaseBETSVoucherRepeater):
+    friendly_name = _("BETS - Chemist Voucher Forwarding (voucher case type)")
+    event_id = CHEMIST_VOUCHER_EVENT
+    voucher_type = "prescription"
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.bets.views import ChemistBETSVoucherRepeaterView
+        return reverse(ChemistBETSVoucherRepeaterView.urlname, args=[domain])
+
+
+class LabBETSVoucherRepeater(BaseBETSVoucherRepeater):
+    friendly_name = _("BETS - Lab Voucher Forwarding (voucher case type)")
+    event_id = LAB_VOUCHER_EVENT
+    voucher_type = "test"
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.bets.views import LabBETSVoucherRepeaterView
+        return reverse(LabBETSVoucherRepeaterView.urlname, args=[domain])
+
+
+class BETS180TreatmentRepeater(BaseBETSRepeater):
+    friendly_name = _(
+        "BETS - MBBS+ Providers: 180 days of private OR govt. "
+        "FDCs with treatment outcome reported (episode case type)"
+    )
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.bets.views import BETS180TreatmentRepeaterView
+        return reverse(BETS180TreatmentRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, episode_case):
+        if not self.case_types_and_users_allowed(episode_case):
+            return False
+
+        case_properties = episode_case.dynamic_case_properties()
+        treatment_outcome = case_properties.get('treatment_outcome', None)
```

Comment:
just to be safe, I think we should also add `case_properties.get(ENROLLED_IN_PRIVATE) == 'true'` here too